python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -381,10 +385,23 @@ private static void generateNames(PythonType resolvedType, List<Name> names, Lis
 
   @Override
   public void visitImportFrom(ImportFrom importFrom) {
-    Optional.of(importFrom)
-      .map(ImportFrom::module)
+    List<String> fromModuleFqn;
+    List<Token> dotPrefixTokens = importFrom.dottedPrefixForModule();
+
+    List<String> dottedPart = Optional.ofNullable(importFrom.module())
       .map(TrivialTypeInferenceVisitor::dottedNameToPartFqn)
-      .ifPresent(fqn -> setTypeToImportFromStatement(importFrom, fqn));
+      .orElse(new ArrayList<>());
+
+    List<String> moduleFqnElements = new ArrayList<>(Arrays.stream(fullyQualifiedModuleName.split("\\.")).toList());
```

Comment:
then this operation makes no sense, but if you prefer to store module name as a single string then this expression could be simpler:
```java
var moduleFqnElements =  List.of(fullyQualifiedModuleName.split("\\."))
```

Reply:
also, does it make sense to define this variable outside of the `if` condition?

Reply:
also if you'll get rid of `dotPrefixTokens` variable and simply assign the `fromModuleFqn` - it will allow to get rid of the `else` part
```java
    var fromModuleFqn = Optional.ofNullable(importFrom.module())
      .map(TrivialTypeInferenceVisitor::dottedNameToPartFqn)
      .orElse(new ArrayList<>());
    var dotPrefixTokens = importFrom.dottedPrefixForModule();
    if (!dotPrefixTokens.isEmpty()) {
      var moduleFqnElements = List.of(fullyQualifiedModuleName.split("\\."));
      // Relative import: we start from the current module FQN and go up as many levels as there are dots in the import statement
      int sizeLimit = Math.max(0, moduleFqnElements.size() - dotPrefixTokens.size());
      fromModuleFqn = Stream.concat(moduleFqnElements.stream().limit(sizeLimit), fromModuleFqn.stream()).toList();
    }
    setTypeToImportFromStatement(importFrom, fromModuleFqn);
```

Reply:
That makes sense, thanks!